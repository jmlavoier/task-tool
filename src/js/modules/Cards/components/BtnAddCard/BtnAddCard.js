import React from 'react';
import PropTypes from 'prop-types';

import style from './BtnAddCard.sass';

const BtnAddCard = ({ onClick }) => (
  <button onClick={onClick} className={style['btn-add-card']} > + Add Card </button>
);

BtnAddCard.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default BtnAddCard;
