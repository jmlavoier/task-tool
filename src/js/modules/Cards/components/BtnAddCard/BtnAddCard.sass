.btn-add-card
  border: 1px solid #c6c8c4
  border-radius: 5px
  background-color: transparent
  height: 50px
  margin: 5px 0 
  width: 100%
  color: #f1f1f1
  cursor: pointer
  outline: none
