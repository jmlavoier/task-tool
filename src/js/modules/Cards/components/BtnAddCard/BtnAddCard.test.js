import React from 'react';
import BtnAddCard from './BtnAddCard';
import { shallow, mount } from 'enzyme';

test('Should component render', () => {
  const wrapper = shallow(<BtnAddCard onClick={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should component call onClick', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<BtnAddCard onClick={mockFn} />);

  wrapper.simulate('click');
  expect(mockFn).toHaveBeenCalled();
})