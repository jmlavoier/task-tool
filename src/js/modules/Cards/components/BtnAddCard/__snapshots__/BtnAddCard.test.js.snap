// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should component render 1`] = `
<button
  className="btn-add-card"
  onClick={[Function]}
>
   + Add Card 
</button>
`;
